File: articles/why-origin-stories-of-algorithms-need-care/prolog/origin_care_score.pl
sum_list([], 0).
sum_list([H|T], Sum) :- sum_list(T, Rest), Sum is H + Rest.

origin_care_score(Scores, Score) :-
    sum_list(Scores, Sum),
    length(Scores, N),
    Score is Sum / N.

% Query example:
% ?- origin_care_score([0.96,0.98,0.96], Score).

File: articles/why-origin-stories-of-algorithms-need-care/fortran/origin_care_score.f90
program origin_care_score_example
  implicit none
  real :: scores(10)
  scores = (/0.96, 0.98, 0.96, 0.88, 0.98, 0.90, 0.90, 0.96, 0.98, 0.98/)
  print *, "origin_care_score=", sum(scores) / size(scores)
end program origin_care_score_example
